reframe/core/systems.py

```diff
@@ -13,6 +13,169 @@
 from reframe.core.environments import (Environment, ProgEnvironment)
 
 
+class Processor(jsonext.JSONSerializable):
+    '''A representation of a processor inside ReFrame.
+
+    .. warning::
+       Users may not create :class:`Processor` objects directly.
+    '''
+
+    def __init__(self, processor_info):
+        self._arch = None
+        self._num_cpus = None
+        self._num_cpus_per_core = None
+        self._num_cpus_per_socket = None
```

Comment:
- `num_cpus_per_core`: lscpu, papi_avail and likwid-topology are using `threads per core` -> can we use `num_threads_per_core` ?

Reply:
If we go with the `threads_per_core` terminology we should name our basic unit as "thread," which I don't like it. "Processing Unit" is the best term or "logical cpu". If this is our basic term, then "num_cpus_per_core" makes sense. I have no problem supporting the "threads_per_core" as an alias, but if we do, we should keep in mind that this also breaks another convention: it doesn't start with "num_". I would see the "threads per core" more like a description term rather than a programmatic one.